Cargo.toml

```diff
@@ -134,11 +134,10 @@ members = [
 	"primitives/transaction-pool",
 	"primitives/trie",
 	"primitives/wasm-interface",
-	"test/utils/chain-spec-builder",
 	"test/utils/client",
 	"test/utils/runtime",
 	"test/utils/runtime/client",
-	"test/utils/transaction-factory",
```

Comment:
Why is that moved to client? It is not related to client. It is a test util (that is mostly broken, but still test).

Reply:
It is not a test-util... The only one using it is actually `node-cli`, pulling it in as normal dependency, using in regular (non-test) builds:
```

[dependencies]
# third-party dependencies
codec = { package = "parity-scale-codec", version = "1.0.6" }
serde = { version = "1.0.102", features = ["derive"] }
futures01 = { package = "futures", version = "0.1.29" }
futures = { version = "0.3.1", features = ["compat"] }
hex-literal = "0.2.1"
jsonrpc-core = "14.0.3"
log = "0.4.8"
rand = "0.7.2"
structopt = "0.3.3"

# primitives
authority-discovery-primitives = { package = "sp-authority-discovery", path = "../../../primitives/authority-discovery"}
babe-primitives = { package = "sp-consensus-babe", path = "../../../primitives/consensus/babe" }
grandpa-primitives = { package = "sp-finality-grandpa", path = "../../../primitives/finality-grandpa" }
primitives = { package = "sp-core", path = "../../../primitives/core" }
sp-runtime = { path = "../../../primitives/runtime" }
sp-timestamp = { path = "../../../primitives/timestamp", default-features = false }
sp-finality-tracker = { path = "../../../primitives/finality-tracker", default-features = false }
inherents = { package = "sp-inherents", path = "../../../primitives/inherents" }
keyring = { package = "sp-keyring", path = "../../../primitives/keyring" }
sp-io = { path = "../../../primitives/io" }
consensus-common = { package = "sp-consensus", path = "../../../primitives/consensus/common" }

# client dependencies
client-api = { package = "sc-client-api", path = "../../../client/api" }
client = { package = "sc-client", path = "../../../client/" }
chain-spec = { package = "sc-chain-spec", path = "../../../client/chain-spec" }
txpool = { package = "sc-transaction-pool", path = "../../../client/transaction-pool" }
txpool-api = { package = "sp-transaction-pool", path = "../../../primitives/transaction-pool" }
network = { package = "sc-network", path = "../../../client/network" }
babe = { package = "sc-consensus-babe", path = "../../../client/consensus/babe" }
grandpa = { package = "sc-finality-grandpa", path = "../../../client/finality-grandpa" }
client-db = { package = "sc-client-db", path = "../../../client/db", default-features = false }
offchain = { package = "sc-offchain", path = "../../../client/offchain" }
sc-rpc = { path = "../../../client/rpc" }
sc-basic-authority = { path = "../../../client/basic-authorship" }
sc-service = { path = "../../../client/service", default-features = false }
sc-telemetry = { path = "../../../client/telemetry" }
authority-discovery = { package = "sc-authority-discovery", path = "../../../client/authority-discovery"}

# frame dependencies
indices = { package = "pallet-indices", path = "../../../frame/indices" }
timestamp = { package = "pallet-timestamp", path = "../../../frame/timestamp", default-features = false }
contracts = { package = "pallet-contracts", path = "../../../frame/contracts" }
system = { package = "frame-system", path = "../../../frame/system" }
balances = { package = "pallet-balances", path = "../../../frame/balances" }
transaction-payment = { package = "pallet-transaction-payment", path = "../../../frame/transaction-payment" }
support = { package = "frame-support", path = "../../../frame/support", default-features = false }
im_online = { package = "pallet-im-online", path = "../../../frame/im-online", default-features = false }
sr-authority-discovery = { package = "pallet-authority-discovery", path = "../../../frame/authority-discovery"}

# node-specific dependencies
node-runtime = { path = "../runtime" }
node-rpc = { path = "../rpc" }
node-primitives = { path = "../primitives" }
node-executor = { path = "../executor" }

# CLI-specific dependencies
tokio = { version = "0.1.22", optional = true }
sc-cli = { path = "../../../client/cli", optional = true }
sc-transaction-factory = { path = "../../../client/transaction-factory", optional = true }
ctrlc = { version = "3.1.3", features = ["termination"], optional = true }

```

Reply:
Yeah, I know that it is pulled in as a normal dependency to `node-cli`. It is not used in tests there, but is more existent to write tests against the runtime from the outside. Which makes it a `test-util`.